pkg/network/usm/postgres_monitor_test.go

```diff
@@ -621,3 +624,155 @@ func validatePostgres(t *testing.T, monitor *Monitor, expectedStats map[string]m
 		return reflect.DeepEqual(expectedStats, found)
 	}, time.Second*5, time.Millisecond*100, "Expected to find a %v stats, instead captured %v", &expectedStats, &found)
 }
+
+func (s *postgresProtocolParsingSuite) TestKernelTelemetry() {
+	t := s.T()
+
+	tests := []struct {
+		name  string
+		isTLS bool
+	}{
+		{
+			name:  "without TLS",
+			isTLS: false,
+		},
+		{
+			name:  "with TLS",
+			isTLS: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.isTLS && !gotlstestutil.GoTLSSupported(t, config.New()) {
+				t.Skip("GoTLS not supported for this setup")
+			}
+			testKernelMessagesCount(t, tt.isTLS)
+		})
+	}
+}
+
+// testKernelMessagesCount check postgres kernel messages count.
+func testKernelMessagesCount(t *testing.T, isTLS bool) {
+	serverHost := "127.0.0.1"
+	serverAddress := net.JoinHostPort(serverHost, postgresPort)
+	require.NoError(t, postgres.RunServer(t, serverHost, postgresPort, isTLS))
+	waitForPostgresServer(t, serverAddress, isTLS)
+
+	monitor := setupUSMTLSMonitor(t, getPostgresDefaultTestConfiguration(isTLS))
+	if isTLS {
+		utils.WaitForProgramsToBeTraced(t, "go-tls", os.Getpid(), utils.ManualTracingFallbackEnabled)
+	}
+	cleanProtocolMaps(t, "postgres_kernel", monitor.ebpfProgram.Manager.Manager)
+
+	context := pgTestContext{
+		serverPort:    postgresPort,
+		targetAddress: serverAddress,
+		serverAddress: serverAddress,
+		extras:        map[string]interface{}{},
+	}
+	expectedMsgCounts := &pgebpf.PostgresKernelMsgCount{
+		Messages_count_buckets: [8]uint64{0, 0, 0, 0, 0, 0, 0, 0},
+	}
+
+	for i := 0; i < pgebpf.KerMsgCountNumBuckets; i++ {
+		test_name := fmt.Sprintf("kernel messages count bucket[%d]", i)
+		t.Run(test_name, func(t *testing.T) {
+
+			t.Cleanup(func() {
+				pgEntry, ok := context.extras["pg"]
+				if !ok {
+					return
+				}
+				pg := pgEntry.(*postgres.PGXClient)
+				defer pg.Close()
+				_ = pg.RunQuery(dropTableQuery)
+			})
+			require.NoError(t, monitor.Pause())
+
+			setupPgClient(t, context, isTLS)
+			require.NoError(t, monitor.Resume())
+
+			obj, ok := context.extras["pg"]
+			require.True(t, ok)
+
+			pgClient := obj.(*postgres.PGXClient)
+			require.NoError(t, pgClient.Ping())
+			// 'sqlInsertCount' is the heuristic number of values to construct the query
+			// test passed for KerMsgCountBucketSize 5 and 10
+			sqlInsertCount := i * pgebpf.KerMsgCountBucketSize
+			runQuesryForBucket(t, context, sqlInsertCount)
+			require.NoError(t, monitor.Pause())
+
+			expectedMsgCounts.Messages_count_buckets[i] = 1
```

Comment:
`runQuesryForBucket(t, context, sqlInsertCount)` runs 3 queries (create, insert, select), and there is no change to the number of messages in a single packet
So I think the test is not working as expected

Reply:
The size of queries is different depending on sqlInsertCount. With bigger counter makes bigger queries and number of Postgres messages increases.

Reply:
These are metrics captured in UI from this test case (see screenshot)
<img width="1422" alt="dddev_metrisc_pg_msg_count" src="https://github.com/user-attachments/assets/3be38822-9d28-4d15-b94d-3dec867de0c9">

Reply:
> These are metrics captured in UI from this test case (see screenshot)

How can metrics be captured from the test?
The test does not communicate with the backend

Reply:
> The size of queries is different depending on sqlInsertCount. With bigger counter makes bigger queries and number of Postgres messages increases.

How can you guarantee that? what forces the split to the exact amount of message?

Reply:
We have no control over how Postgres splits requests into internal messages, we can only expect larger requests to be broken up into more messages. We can only check that larger queries create more messages and thus fill different buckets.

Reply:
The test only checks whether all bucket are filling. The Postgres splits queries into messages and may request different TCP packet size from OS. We cannot predict exact number of messages in each bucket.

Reply:
I refactored test program